### PR TITLE
Add 'toString' method into IdleStateEvent

### PR DESCRIPTION
**Pull request by ursaj**
### Motivation:

IdleStateEvent is very convenient and frequently used type of events. However both in runtime (logs) and in debug you need some manual steps to see their actual content. Default implementation generates worthless trash like this:

    io.netty.handler.timeout.IdleStateEvent@27f674d

There are examples already, where event has convenient and useful toString implementation:

* io.netty.handler.proxy.ProxyConnectionEvent
* io.netty.handler.ssl.SslCompletionEvent

### Modification:

* Implement 'IdleStateEvent.toString' method.
* Unit test.

### CODE REVIEW

File: handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java [hunk 1]
```diff
@@ -31,6 +32,7 @@
 
     private final IdleState state;
     private final boolean first;
+    private String strVal;
```
Comment (carl-mastrangelo) [inline handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java]: You probably don't need to cache this, since it's unlikely to be reused.

Comment (normanmaurer) [reply handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java]: I agree with @carl-mastrangelo ... @ursaj can you remove this  ?

Comment (ursaj) [reply handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java]: you both are mistaken in the way, how this class is used. there are only 6 instances of this class in runtime:

    public class IdleStateEvent {
        public static final IdleStateEvent FIRST_READER_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.READER_IDLE, true);
        public static final IdleStateEvent READER_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.READER_IDLE, false);
        public static final IdleStateEvent FIRST_WRITER_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.WRITER_IDLE, true);
        public static final IdleStateEvent WRITER_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.WRITER_IDLE, false);
        public static final IdleStateEvent FIRST_ALL_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.ALL_IDLE, true);
        public static final IdleStateEvent ALL_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.ALL_IDLE, false);

So cached string value is used countless times.

Comment (ursaj) [reply handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java]: The same applies to inherited instances (if any):

* either they are used in the same shared model, so this optimisation is useful;
* or they are created, used and GC-ed in a short while, so there is no harm in this optimisation.

File: handler/src/test/java/io/netty/handler/timeout/IdleStateEventTest.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package io.netty.handler.timeout;
```
Comment (normanmaurer) [inline handler/src/test/java/io/netty/handler/timeout/IdleStateEventTest.java]: missing license header.

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@ursaj thanks a lot!

File: handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java (hunk 1 above)
Comment (carl-mastrangelo) [reply handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java]: I disagree here.   If it is the case that the same idle events are reused heavily, then it stands to reason that they are shared by multiple event loops.   That makes this code racy and will trigger the race detector.

Comment (ursaj) [reply handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java]: @carl-mastrangelo generate strings on each invocation doesn't look good either. any ideas or examples how it should look like?

Comment (ursaj) [reply handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java]: ah, I see. you've already fixed it in PR #9705

**Pull request merged**